### ISSUE TITLE
dogfood: installinator downloads over the bootstrap network were extremely sluggish / repeatedly timed out

### ISSUE THREAD

**Issue by jgallagher** [oxidecomputer/omicron#3105]
On our second attempt to update a couple gimlets on the dogfood rack today, we appeared to get effectively stuck at the "Downloading host phase 2 artifact" stage in wicket: it stayed at 0%, but when we looked at more details we did see the progress / amount downloaded _very slowly_ ticking up (where by "very slowly", I mean "a few tens of KiB every 15-20 seconds"). I'm attaching the full installinator service log from the gimlet being mupdated below; some notes on it follow.

`[fdb0:a840:2504:11a::2]:12227` is the address of wicketd. Every connection to this address fails with a `body error: operation timed out`; e.g.,

```
Dec 28 00:01:38.878 WARN error after 42.005578677s: peer [fdb0:a840:2504:11a::2]:12227 returned an HTTP error
Caused by:
  -> HTTP client error
  -> Communication Error: request or response body error: operation timed out
  -> request or response body error: operation timed out
  -> operation timed out, remaining_peers: 4, artifact_hash_id: ArtifactHashId { kind: ArtifactKind("host_phase_2"), hash: ArtifactHash("19698dfbd3968947f49137ceabcc2a017aba6ceb340230de74a4750bf8b19d7d") }, component: Peers, artifact: ArtifactHashId { kind: ArtifactKind("host_phase_2"), hash: ArtifactHash("19698dfbd3968947f49137ceabcc2a017aba6ceb340230de74a4750bf8b19d7d") }
```

None of the remaining peers were running servers on port 12227, but we see a mix of "connection refused"; e.g.,

```
Dec 28 00:00:41.871 WARN error after 813.929µs: peer [fdb0:a840:2504:3d4::1]:12227 returned an HTTP error
Caused by:
  -> HTTP client error
  -> Communication Error: error sending request for url (http://[fdb0:a840:2504:3d4::1]:12227/artifacts/by-hash/host_phase_2/19698dfbd3968947f49137ceabcc2a017aba6ceb340230de74a4750bf8b19d7d): error trying to connect: tcp connect error: Connection refused (os error 146)
  -> error sending request for url (http://[fdb0:a840:2504:3d4::1]:12227/artifacts/by-hash/host_phase_2/19698dfbd3968947f49137ceabcc2a017aba6ceb340230de74a4750bf8b19d7d): error trying to connect: tcp connect error: Connection refused (os error 146)
  -> error trying to connect: tcp connect error: Connection refused (os error 146)
  -> tcp connect error: Connection refused (os error 146)
  -> Connection refused (os error 146), remaining_peers: 7, artifact_hash_id: ArtifactHashId { kind: ArtifactKind("host_phase_2"), hash: ArtifactHash("19698dfbd3968947f49137ceabcc2a017aba6ceb340230de74a4750bf8b19d7d") }, component: Peers, artifact: ArtifactHashId { kind: ArtifactKind("host_phase_2"), hash: ArtifactHash("19698dfbd3968947f49137ceabcc2a017aba6ceb340230de74a4750bf8b19d7d") }
```

(what I'd expect if we try to connect to a system that isn't running a server) and connection timeouts; e.g.,

```
Dec 28 00:01:53.880 WARN error after 15.002245711s: peer [fdb0:a840:2504:191::1]:12227 returned an HTTP error
Caused by:
  -> HTTP client error
  -> Communication Error: error sending request for url (http://[fdb0:a840:2504:191::1]:12227/artifacts/by-hash/host_phase_2/19698dfbd3968947f49137ceabcc2a017aba6ceb340230de74a4750bf8b19d7d): operation timed out
  -> error sending request for url (http://[fdb0:a840:2504:191::1]:12227/artifacts/by-hash/host_phase_2/19698dfbd3968947f49137ceabcc2a017aba6ceb340230de74a4750bf8b19d7d): operation timed out
  -> operation timed out, remaining_peers: 3, artifact_hash_id: ArtifactHashId { kind: ArtifactKind("host_phase_2"), hash: ArtifactHash("19698dfbd3968947f49137ceabcc2a017aba6ceb340230de74a4750bf8b19d7d") }, component: Peers, artifact: ArtifactHashId { kind: ArtifactKind("host_phase_2"), hash: ArtifactHash("19698dfbd3968947f49137ceabcc2a017aba6ceb340230de74a4750bf8b19d7d") }
```

(what I'd expect if this peer is not responding to TCP request at all?). I currently have no explanation for

a) why installinator -> wicketd was timing out every time
b) how installinator was still claiming to make (extremely slow) forward progress

To add to the confusion, two other data points for the being-mupdated gimlet in question:

a) The SP console via faux-mgs appeared to be unresponsive
b) When I ssh'd into the gimlet via its bootstrap address, the ssh connection seemed perfectly healthy, and I was able to manually fetch the host artifact from wicketd using curl (using the same `[fdb0:a840:2504:11a::2]:12227` address installinator was trying to use) with no problems at all.

[oxide-installinator:default.log](https://github.com/oxidecomputer/omicron/files/11467369/oxide-installinator.default.log)

**Comment by rcgoodfellow** [comment]
Do you happen to know what the MTU values were on the machines in question?

**Comment by leftwo** [comment]
I had set all `cxgbe` interfaces to have 1600 initially, but if the hosts rebooted then I'm pretty sure they would
have defaulted back to 9000.

**Comment by jgallagher** [comment]
This was fixed by #3159
